arkouda/dataframe.py

```diff
@@ -382,25 +382,83 @@ def _repr_html_(self):
     def _ipython_key_completions_(self):
         return self._columns
 
-    def drop(self, keys):
+    def _drop_column(self, keys):
         """
         Drop a column or columns from the dataframe, in-place.
 
-        Parameters
-        ----------
-        keys : str or list
-            The column(s) to be dropped.
+        keys : list
+            The labels to be dropped on the given axis
         """
-
-        if isinstance(keys, str):
-            keys = [keys]
         for key in keys:
             # Do not allow the user to drop the index column
             if key == 'index':
                 raise KeyError('The index column may be reset, but not dropped.')
 
             del self[key]
 
+    def _drop_row(self, keys):
+        """
+        Drop a row or rows from the dataframe, in-place.
+
+        keys : list
+            The indexes to be dropped on the given axis
+        """
+        idx_list = []
+        last_idx = -1
+        # sort to ensure we go in ascending order.
+        keys.sort()
+        for i, k in enumerate(keys):
+            if not isinstance(k, int):
+                raise TypeError("Index keys must be integers.")
+            pda = self.index[(last_idx+1):k:1]
+            if pda.size > 0:
+                idx_list.append(pda)
+            last_idx = k
+
+        pda = self.index[(last_idx+1):]
+        if pda.size > 0:
+            idx_list.append(pda)
```

Comment:
I just chatted with ethan about this offline. but if the `pda.size >0` check isn't necessary (i.e. `concatenate([],...)` behaves the way we want it to with an empty list), then we can simplify the logic here and avoid creating the extra `pda` with something like

```python
idx_list.append(self.index[(last_idx+1):k])
# and
idx_list.append(self.index[(last_idx+1):])

Reply:
Removed the `if pda.size > 0` and updated append to those listed above.